From 9d55b26a7ab60556141300cac68b3205d1f732c5 Mon Sep 17 00:00:00 2001
From: Alberto Milone <alberto.milone@canonical.com>
Date: Mon, 4 Nov 2013 20:21:09 +0100
Subject: [PATCH 1/1] Add support for Linux 3.12

The uid_t that we need is now a struct member of the kuid_t that
current_euid() returns.
---
 firegl_public.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/firegl_public.c b/firegl_public.c
index 890a0aa..f1fa72c 100755
--- a/firegl_public.c
+++ b/firegl_public.c
@@ -1740,7 +1740,12 @@ KCL_TYPE_Pid ATI_API_CALL KCL_GetTgid(void)
 KCL_TYPE_Uid ATI_API_CALL KCL_GetEffectiveUid(void)
 {
 #ifdef current_euid
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,12,0)
+    kuid_t tmp_kuid = current_euid();
+    return tmp_kuid.val;
+#else
     return current_euid();
+#endif
 #else
     return current->euid;
 #endif
-- 
1.7.9.5

